From bf6f7990fd38dec8d6cb6359b931483107909db5 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <ondrej.jirman@zonio.net>
Date: Tue, 13 Mar 2007 02:08:59 +0100
Subject: [PATCH] use password manager key from ESource auth-key property in ECal

---
 calendar/libecal/e-cal.c |   33 ++++++++++++++++++---------------
 1 files changed, 18 insertions(+), 15 deletions(-)

diff --git a/calendar/libecal/e-cal.c b/calendar/libecal/e-cal.c
index f506084..264ef2d 100644
--- a/calendar/libecal/e-cal.c
+++ b/calendar/libecal/e-cal.c
@@ -1747,21 +1747,24 @@ open_calendar (ECal *ecal, gboolean only_if_exists, GError **error, ECalendarSta
 		prompt = g_strdup_printf (_("Enter password for %s (user %s)"),
 				e_source_peek_name (priv->source), username);
 
-		auth_type = e_source_get_property (priv->source, "auth-type");
-		if (auth_type) 
-			key = build_pass_key (ecal);
-		else {
-			parent_user = e_source_get_property (priv->source, "parent_id_name");
-			if (parent_user) {
-				key = build_proxy_pass_key (ecal, parent_user);
-				/* 
-				   This password prompt will be prompted rarely. Since the key that is passed to 
-				   the auth_func corresponds to the parent user.
-				 */
-				prompt = g_strdup_printf (_("Enter password for %s to enable proxy for user %s"), e_source_peek_name (priv->source), parent_user);
-			} else 
-				key = g_strdup (e_cal_get_uri (ecal));
-		}
+		key = g_strdup(e_source_get_property (priv->source, "auth-key"));
+		if (!key) {
+      auth_type = e_source_get_property (priv->source, "auth-type");
+      if (auth_type) 
+        key = build_pass_key (ecal);
+      else {
+        parent_user = e_source_get_property (priv->source, "parent_id_name");
+        if (parent_user) {
+          key = build_proxy_pass_key (ecal, parent_user);
+          /* 
+             This password prompt will be prompted rarely. Since the key that is passed to 
+             the auth_func corresponds to the parent user.
+           */
+          prompt = g_strdup_printf (_("Enter password for %s to enable proxy for user %s"), e_source_peek_name (priv->source), parent_user);
+        } else 
+          key = g_strdup (e_cal_get_uri (ecal));
+      }
+    }
 
 		if (!key) {
 			e_calendar_remove_op (ecal, our_op);
-- 
1.5.1.1.GIT-dirty

