diff --git a/calendar/gui/dialogs/send-comp.c b/calendar/gui/dialogs/send-comp.c
index 11d6e74..07c0bd3 100644
--- a/calendar/gui/dialogs/send-comp.c
+++ b/calendar/gui/dialogs/send-comp.c
@@ -42,6 +42,11 @@
 gboolean
 send_component_dialog (GtkWindow *parent, ECal *client, ECalComponent *comp, gboolean new)
 {
+  if (g_str_has_prefix(e_cal_get_uri(client), "eee://"))
+  {
+    return FALSE;
+  }
+
 	ECalComponentVType vtype;
 	const char *id;
