From e5e3b24f24fa8ab33dc0830b270ff36b9772a8c4 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <ondrej.jirman@zonio.net>
Date: Sat, 2 Jun 2007 18:19:23 +0200
Subject: [PATCH] Fix ESource property race in e-cal.c:open_calendar().

---
 calendar/libecal/e-cal.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/calendar/libecal/e-cal.c b/calendar/libecal/e-cal.c
index b69e47b..9f32d85 100644
--- a/calendar/libecal/e-cal.c
+++ b/calendar/libecal/e-cal.c
@@ -1734,7 +1734,7 @@ open_calendar (ECal *ecal, gboolean only_if_exists, GError **error, ECalendarSta
 			E_CALENDAR_CHECK_STATUS (E_CALENDAR_STATUS_AUTHENTICATION_REQUIRED, error);
 		}
 
-		username = e_source_get_property (priv->source, "username");
+		username = g_strdup(e_source_get_property (priv->source, "username"));
 		if (!username) {
 			e_calendar_remove_op (ecal, our_op);
 			g_mutex_unlock (our_op->mutex);
@@ -1799,8 +1799,8 @@ open_calendar (ECal *ecal, gboolean only_if_exists, GError **error, ECalendarSta
 					   username ? username : "",
 					   password ? password : "",
 					   &ev);
-	if (password)
-		g_free (password);
+  g_free (password);
+  g_free (username);
 
 	if (BONOBO_EX (&ev)) {
 		e_calendar_remove_op (ecal, our_op);
-- 
1.5.2.GIT-dirty

