From 2d1e7d589504d5f94a3fd7bb342a097656c760fd Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <ondrej.jirman@zonio.net>
Date: Sat, 23 Jun 2007 13:58:04 +0200
Subject: [PATCH] expand new groups by default in ESourceSelector

---
 libedataserverui/e-source-selector.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/libedataserverui/e-source-selector.c b/libedataserverui/e-source-selector.c
index ed861e7..f94eb11 100644
--- a/libedataserverui/e-source-selector.c
+++ b/libedataserverui/e-source-selector.c
@@ -253,11 +253,14 @@ rebuild_model (ESourceSelector *selector)
 		ESourceGroup *group = E_SOURCE_GROUP (p->data);
 		GSList *sources, *q;
 		GtkTreeRowReference *row_ref;
+		GtkTreePath *path_expand = NULL;
 		
 		row_ref = g_hash_table_lookup (rebuild_data->remaining_uids, e_source_group_peek_uid (group));
 		if (!row_ref) {
+
 			gtk_tree_store_append (GTK_TREE_STORE (tree_store), &iter, NULL);
 			gtk_tree_store_set (GTK_TREE_STORE (tree_store), &iter, 0, group, -1);
+			path_expand = gtk_tree_model_get_path (GTK_TREE_MODEL (tree_store), &iter);
 		} else {
 			GtkTreePath *path;
 			
@@ -294,6 +297,10 @@ rebuild_model (ESourceSelector *selector)
 				gtk_tree_path_free (path);
 			}
 		}
+		if (path_expand) {
+			gtk_tree_view_expand_row (GTK_TREE_VIEW (selector), path_expand, TRUE);
+			gtk_tree_path_free (path_expand);
+		}
 	}
 
 	if (rebuild_data->selection_changed)
-- 
1.5.4-dirty

