From fd4788f22a6f8d646aa90fcbef1a7eb5b07ede65 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <ondrej.jirman@zonio.net>
Date: Sun, 14 Oct 2007 20:24:34 +0200
Subject: [PATCH] remove useless EBook access from EMeetingStore

---
 calendar/gui/e-meeting-store.c |   15 ---------------
 1 files changed, 0 insertions(+), 15 deletions(-)

diff --git a/calendar/gui/e-meeting-store.c b/calendar/gui/e-meeting-store.c
index 1c3b645..65a4930 100644
--- a/calendar/gui/e-meeting-store.c
+++ b/calendar/gui/e-meeting-store.c
@@ -29,7 +29,6 @@
 #include <glib/gi18n.h>
 #include <libgnome/gnome-util.h>
 #include <libgnomevfs/gnome-vfs.h>
-#include <libebook/e-book.h>
 #include <libecal/e-cal-component.h>
 #include <libecal/e-cal-util.h>
 #include <libecal/e-cal-time-util.h>
@@ -51,8 +50,6 @@ struct _EMeetingStorePrivate {
 	
 	char *fb_uri;
 
-	EBook *ebook;
-
 	GPtrArray *refresh_queue;
 	GHashTable *refresh_data;
 	GMutex *mutex;
@@ -86,13 +83,6 @@ static GObjectClass *parent_class = NULL;
 
 static void start_async_read (GnomeVFSAsyncHandle *handle, GnomeVFSResult result, gpointer data);
 
-static void
-start_addressbook_server (EMeetingStore *store)
-{
-	store->priv->ebook = e_book_new_system_addressbook (NULL);
-	e_book_open (store->priv->ebook, FALSE, NULL);
-}
-
 static icalparameter_cutype
 text_to_type (const char *type)
 {
@@ -553,9 +543,6 @@ ems_finalize (GObject *obj)
 	if (priv->client != NULL)
 		g_object_unref (priv->client);
 
-	if (priv->ebook != NULL)
-		g_object_unref (priv->ebook);
-
  	while (priv->refresh_queue->len > 0)
  		refresh_queue_remove (store, g_ptr_array_index (priv->refresh_queue, 0));
  	g_ptr_array_free (priv->refresh_queue, TRUE);
@@ -604,8 +591,6 @@ ems_init (EMeetingStore *store)
 	priv->mutex = g_mutex_new ();
 
 	priv->num_queries = 0;
-
-	start_addressbook_server (store);
 }
 
 GType
-- 
1.5.3.4.320-dirty

