From ba483f485c9bec1d9b9eea0b7a850b93ec2f8a6a Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <ondrej.jirman@zonio.net>
Date: Sun, 9 Mar 2008 13:30:37 +0100
Subject: [PATCH] fix address exclusion in comp_to_list()

---
 calendar/gui/itip-utils.c |   24 +++++++++++++++++-------
 1 files changed, 17 insertions(+), 7 deletions(-)

diff --git a/calendar/gui/itip-utils.c b/calendar/gui/itip-utils.c
index 173817f..f12874b 100644
--- a/calendar/gui/itip-utils.c
+++ b/calendar/gui/itip-utils.c
@@ -557,6 +557,8 @@ comp_to_list (ECalComponentItipMethod method, ECalComponent *comp, GList *users,
 			for (l = attendees; l != NULL; l = l->next) {
 				ECalComponentAttendee *att = l->data;
 
+				if (users_has_attendee (users, att->value))
+					continue;
 
 				recipient = &(to_list->_buffer[to_list->_length]);
 				if (att->cn)
@@ -582,6 +584,8 @@ comp_to_list (ECalComponentItipMethod method, ECalComponent *comp, GList *users,
 
 			e_cal_component_get_organizer (comp, &organizer);
 			if (organizer.value) {
+				if (users_has_attendee (users, organizer.value))
+					return to_list;
 				recipient->name = CORBA_string_dup ("");
 				recipient->address = CORBA_string_dup (itip_strip_mailto (organizer.value));
 				to_list->_length++;
@@ -612,15 +616,18 @@ comp_to_list (ECalComponentItipMethod method, ECalComponent *comp, GList *users,
 		to_list->_length = 0;
 		to_list->_buffer = CORBA_sequence_GNOME_Evolution_Composer_Recipient_allocbuf (len);
 		recipient = &(to_list->_buffer[0]);
-		to_list->_length++;
 
-		if (organizer.cn != NULL)
-			recipient->name = CORBA_string_dup (organizer.cn);
-		else
-			recipient->name = CORBA_string_dup ("");
-		recipient->address = CORBA_string_dup (itip_strip_mailto (organizer.value));
+		if (!users_has_attendee (users, organizer.value)) {
+			to_list->_length++;
 
-		/* send the status to delegatee to the delegate also*/
+			if (organizer.cn != NULL)
+				recipient->name = CORBA_string_dup (organizer.cn);
+			else
+				recipient->name = CORBA_string_dup ("");
+			recipient->address = CORBA_string_dup (itip_strip_mailto (organizer.value));
+		}
+		
+		/* send the status to delegatee to the delegate also*/	
 		e_cal_component_get_attendee_list (comp, &attendees);
 		sender = itip_get_comp_attendee (comp, NULL);
 
@@ -632,6 +639,9 @@ comp_to_list (ECalComponentItipMethod method, ECalComponent *comp, GList *users,
 				if (!(att->delfrom && *att->delfrom))
 					break;
 
+				if (users_has_attendee (users, att->value))
+					continue;
+
 				recipient = &(to_list->_buffer[to_list->_length]);
 				recipient->name = CORBA_string_dup ("");
 				recipient->address = CORBA_string_dup (itip_strip_mailto (att->delfrom));
-- 
1.5.4-dirty

