diff --git a/plugins/itip-formatter/itip-formatter.c b/plugins/itip-formatter/itip-formatter.c
index b468f6c..0b2ab2e 100644
--- a/plugins/itip-formatter/itip-formatter.c
+++ b/plugins/itip-formatter/itip-formatter.c
@@ -1590,6 +1590,24 @@ view_response_cb (GtkWidget *widget, ItipViewResponse response, gpointer data)
 		icalcomponent_add_property (pitip->ical_comp, prop);
 	}
 
+    if (g_str_has_prefix(e_cal_get_uri(pitip->current_ecal), "eee://"))
+    {
+        const gchar *comment = NULL;
+       comment = itip_view_get_rsvp_comment (ITIP_VIEW (pitip->view));
+       if (comment) {
+               GSList comments;
+               ECalComponentText text;
+    
+               text.value = comment;
+               text.altrep = NULL;
+    
+               comments.data = &text;
+               comments.next = NULL;
+    
+               e_cal_component_set_comment_list (pitip->comp, &comments);
+       }
+    }
+
 	switch (response) {
 		case ITIP_VIEW_RESPONSE_ACCEPT:
 			if (pitip->type != E_CAL_SOURCE_TYPE_JOURNAL)
