From 949562976b37d4cd6798f66ff7f16780b48a0ede Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <ondrej.jirman@zonio.net>
Date: Sat, 27 Oct 2007 07:33:44 +0200
Subject: [PATCH] show error message in event tooltip

---
 calendar/gui/e-calendar-view.c |   15 +++++++++++++++
 1 files changed, 15 insertions(+), 0 deletions(-)

diff --git a/calendar/gui/e-calendar-view.c b/calendar/gui/e-calendar-view.c
index f52d05d..f1801b9 100644
--- a/calendar/gui/e-calendar-view.c
+++ b/calendar/gui/e-calendar-view.c
@@ -2189,6 +2189,20 @@ e_calendar_view_get_tooltips (ECalendarViewEventData *data)
 	gtk_box_pack_start ((GtkBox *)box, ebox, FALSE, FALSE, 0);
 	g_free (tmp);
 	
+  /* display error */
+  e_cal_component_get_x_property (newcomp, "X-EVOLUTION-ERROR", &str);
+	if (str) {
+		tmp = g_strdup_printf (_("Error: %s"), str);
+		label = gtk_label_new (NULL);
+		gtk_label_set_markup ((GtkLabel *)label, tmp);
+		hbox = gtk_hbox_new (FALSE, 0);
+		gtk_box_pack_start ((GtkBox *)hbox, label, FALSE, FALSE, 0);
+		ebox = gtk_event_box_new ();
+		gtk_container_add ((GtkContainer *)ebox, hbox);
+		gtk_box_pack_start ((GtkBox *)box, ebox, FALSE, FALSE, 0);
+		g_free (tmp);
+	}
+
 	e_cal_component_get_organizer (newcomp, &organiser);
 	if (organiser.cn) {
 		char *ptr ; 
@@ -2227,6 +2241,7 @@ e_calendar_view_get_tooltips (ECalendarViewEventData *data)
 		gtk_box_pack_start ((GtkBox *)box, ebox, FALSE, FALSE, 0);
 		g_free (tmp);
 	}
+
 	e_cal_component_get_dtstart (newcomp, &dtstart);
 	e_cal_component_get_dtend (newcomp, &dtend);
 			
-- 
1.5.3.4.320-dirty

