From 4ce4c66c80bd820639476aa14e42689be2fbd05d Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <ondrej.jirman@zonio.net>
Date: Sun, 9 Mar 2008 13:30:37 +0100
Subject: [PATCH] fix address exclusion in comp_to_list()

---
 calendar/gui/itip-utils.c |   22 ++++++++++++++++------
 1 files changed, 16 insertions(+), 6 deletions(-)

diff --git a/calendar/gui/itip-utils.c b/calendar/gui/itip-utils.c
index 2cfd524..5c5f310 100644
--- a/calendar/gui/itip-utils.c
+++ b/calendar/gui/itip-utils.c
@@ -559,6 +559,8 @@ comp_to_list (ECalComponentItipMethod method, ECalComponent *comp, GList *users,
 			for (l = attendees; l != NULL; l = l->next) {
 				ECalComponentAttendee *att = l->data;
 
+				if (users_has_attendee (users, att->value))
+					continue;
 			
 				recipient = &(to_list->_buffer[to_list->_length]);
 				if (att->cn)
@@ -584,6 +586,8 @@ comp_to_list (ECalComponentItipMethod method, ECalComponent *comp, GList *users,
 			
 			e_cal_component_get_organizer (comp, &organizer);
 			if (organizer.value) {
+				if (users_has_attendee (users, organizer.value))
+					return to_list;
 				recipient->name = CORBA_string_dup ("");
 				recipient->address = CORBA_string_dup (itip_strip_mailto (organizer.value));
 				to_list->_length++;
@@ -614,13 +618,16 @@ comp_to_list (ECalComponentItipMethod method, ECalComponent *comp, GList *users,
 		to_list->_length = 0;
 		to_list->_buffer = CORBA_sequence_GNOME_Evolution_Composer_Recipient_allocbuf (len);
 		recipient = &(to_list->_buffer[0]);
-		to_list->_length++;
 
-		if (organizer.cn != NULL)
-			recipient->name = CORBA_string_dup (organizer.cn);
-		else
-			recipient->name = CORBA_string_dup ("");
-		recipient->address = CORBA_string_dup (itip_strip_mailto (organizer.value));
+		if (!users_has_attendee (users, organizer.value)) {
+			to_list->_length++;
+
+			if (organizer.cn != NULL)
+				recipient->name = CORBA_string_dup (organizer.cn);
+			else
+				recipient->name = CORBA_string_dup ("");
+			recipient->address = CORBA_string_dup (itip_strip_mailto (organizer.value));
+		}
 		
 		/* send the status to delegatee to the delegate also*/	
 		e_cal_component_get_attendee_list (comp, &attendees);
@@ -634,6 +641,9 @@ comp_to_list (ECalComponentItipMethod method, ECalComponent *comp, GList *users,
 				if (!(att->delfrom && *att->delfrom))
 					break;
 
+				if (users_has_attendee (users, att->value))
+					continue;
+
 				recipient = &(to_list->_buffer[to_list->_length]);
 				recipient->name = CORBA_string_dup ("");
 				recipient->address = CORBA_string_dup (itip_strip_mailto (att->delfrom));
-- 
1.5.4-dirty

