diff -Naur a/calendar/alarm-notify/alarm-notify.c b/calendar/alarm-notify/alarm-notify.c
--- a/calendar/alarm-notify/alarm-notify.c	2012-02-21 09:12:53.000000000 +0100
+++ b/calendar/alarm-notify/alarm-notify.c	2012-06-25 01:12:25.241807699 +0200
@@ -485,7 +485,9 @@
 	priv = an->priv;
 	str_uri = e_source_get_uri (source);
 	e_uri = e_uri_new (str_uri);
-	if (e_source_get_property (source, "auth-type"))
+	if (e_source_get_property (source, "auth-key"))
+		pass_key = e_source_get_duped_property (source, "auth-key");
+	else if (e_source_get_property (source, "auth-type"))
 		pass_key = e_uri_to_string (e_uri, FALSE);
 	else
 		pass_key = g_strdup (str_uri);
diff -Naur a/calendar/gui/dialogs/calendar-setup.c b/calendar/gui/dialogs/calendar-setup.c
--- a/calendar/gui/dialogs/calendar-setup.c	2012-03-05 10:31:14.000000000 +0100
+++ b/calendar/gui/dialogs/calendar-setup.c	2012-06-25 01:13:54.645687328 +0200
@@ -78,9 +78,13 @@
 eccp_commit (EConfig *ec,
              CalendarSourceDialog *sdialog)
 {
+	ECalConfigTargetSource *target = (ECalConfigTargetSource *) ec->target;
 	xmlNodePtr xml;
 	GtkWindow *window;
 
+	if (target->disable_source_update)
+		return;
+
 	if (sdialog->original_source) {
 		const gchar *color_spec;
 
diff -Naur a/calendar/gui/dialogs/cancel-comp.c b/calendar/gui/dialogs/cancel-comp.c
--- a/calendar/gui/dialogs/cancel-comp.c	2012-03-05 10:31:14.000000000 +0100
+++ b/calendar/gui/dialogs/cancel-comp.c	2012-06-25 01:14:26.308291492 +0200
@@ -71,6 +71,9 @@
 	ECalComponentVType vtype;
 	const gchar *id;
 
+	if (g_str_has_prefix(e_client_get_uri(cal_client), "eee://"))
+		return FALSE;
+
 	if (deleting && e_cal_client_check_save_schedules (cal_client))
 		return TRUE;
 
diff -Naur a/calendar/gui/dialogs/comp-editor.c b/calendar/gui/dialogs/comp-editor.c
--- a/calendar/gui/dialogs/comp-editor.c	2012-03-05 10:31:14.000000000 +0100
+++ b/calendar/gui/dialogs/comp-editor.c	2012-06-25 01:15:53.056333946 +0200
@@ -3248,7 +3248,9 @@
 
 	if (!e_cal_component_has_attachments (priv->comp) ||
 		e_client_check_capability (E_CLIENT (priv->cal_client),
-		CAL_STATIC_CAPABILITY_CREATE_MESSAGES)) {
+		CAL_STATIC_CAPABILITY_CREATE_MESSAGES) ||
+		e_client_check_capability (E_CLIENT (priv->cal_client),
+		CAL_STATIC_CAPABILITY_NO_SEND_IMIP)) {
 		if (itip_send_comp (
 			method, send_comp, priv->cal_client,
 			NULL, NULL, users, strip_alarms,
diff -Naur a/calendar/gui/dialogs/send-comp.c b/calendar/gui/dialogs/send-comp.c
--- a/calendar/gui/dialogs/send-comp.c	2012-03-05 10:31:14.000000000 +0100
+++ b/calendar/gui/dialogs/send-comp.c	2012-06-25 01:16:48.023334076 +0200
@@ -145,6 +145,9 @@
 	GtkWidget *content_area;
 	gboolean res;
 
+	if (g_str_has_prefix(e_client_get_uri(client), "eee://"))
+		return FALSE;
+
 	if (strip_alarms)
 		*strip_alarms = TRUE;
 
diff -Naur a/calendar/gui/e-cal-config.h b/calendar/gui/e-cal-config.h
--- a/calendar/gui/e-cal-config.h	2012-02-21 09:12:53.000000000 +0100
+++ b/calendar/gui/e-cal-config.h	2012-06-25 01:17:25.814861670 +0200
@@ -75,6 +75,7 @@
 
 	ESource *source;
         ECalClientSourceType source_type;
+	gboolean disable_source_update;
 };
 
 struct _ECalConfigTargetPrefs {
diff -Naur a/calendar/gui/e-calendar-view.c b/calendar/gui/e-calendar-view.c
--- a/calendar/gui/e-calendar-view.c	2012-04-09 14:44:38.000000000 +0200
+++ b/calendar/gui/e-calendar-view.c	2012-06-25 01:18:53.676216005 +0200
@@ -1962,6 +1962,20 @@
 	gtk_box_pack_start ((GtkBox *) box, ebox, FALSE, FALSE, 0);
 	g_free (tmp);
 
+	/* display error */
+	e_cal_component_get_x_property (newcomp, "X-EVOLUTION-ERROR", &str);
+	if (str) {
+		tmp = g_strdup_printf (_("Error: %s"), str);
+		label = gtk_label_new (NULL);
+		gtk_label_set_markup ((GtkLabel *)label, tmp);
+		hbox = gtk_hbox_new (FALSE, 0);
+		gtk_box_pack_start ((GtkBox *)hbox, label, FALSE, FALSE, 0);
+		ebox = gtk_event_box_new ();
+		gtk_container_add ((GtkContainer *)ebox, hbox);
+		gtk_box_pack_start ((GtkBox *)box, ebox, FALSE, FALSE, 0);
+		g_free (tmp);
+	}
+
 	e_cal_component_get_organizer (newcomp, &organiser);
 	if (organiser.cn) {
 		gchar *ptr;
diff -Naur a/calendar/gui/itip-utils.c b/calendar/gui/itip-utils.c
--- a/calendar/gui/itip-utils.c	2012-04-09 14:44:38.000000000 +0200
+++ b/calendar/gui/itip-utils.c	2012-06-25 01:19:32.483730847 +0200
@@ -1553,7 +1553,7 @@
 		only_new_attendees ?  g_object_get_data (
 		G_OBJECT (send_comp), "new-attendees") : NULL);
 	if (method != E_CAL_COMPONENT_METHOD_PUBLISH) {
-		if (destinations == NULL) {
+		if (destinations == NULL || e_client_check_capability (E_CLIENT (cal_client), CAL_STATIC_CAPABILITY_NO_SEND_IMIP)) {
 			/* We sent them all via the server */
 			retval = TRUE;
 			goto cleanup;
diff -Naur a/plugins/itip-formatter/itip-formatter.c b/plugins/itip-formatter/itip-formatter.c
--- a/plugins/itip-formatter/itip-formatter.c	2012-05-14 06:35:26.000000000 +0200
+++ b/plugins/itip-formatter/itip-formatter.c	2012-06-25 01:09:55.346431862 +0200
@@ -2577,6 +2577,25 @@
 		icalcomponent_add_property (pitip->ical_comp, prop);
 	}
 
+	if (g_str_has_prefix(e_client_get_uri(pitip->current_client), "eee://"))
+	{
+		const gchar *comment = NULL;
+		comment = itip_view_get_rsvp_comment (ITIP_VIEW (pitip->view));
+
+		if (comment) {
+			GSList comments;
+			ECalComponentText text;
+
+			text.value = comment;
+			text.altrep = NULL;
+
+			comments.data = &text;
+			comments.next = NULL;
+
+			e_cal_component_set_comment_list (pitip->comp, &comments);
+		}
+	}
+
 	switch (response) {
 		case ITIP_VIEW_RESPONSE_ACCEPT:
 			if (pitip->type != E_CAL_CLIENT_SOURCE_TYPE_MEMOS)
