From 1fd57b328a8a522c1394cdd00f8843880263d2f5 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <ondrej.jirman@zonio.net>
Date: Tue, 13 Mar 2007 02:08:59 +0100
Subject: [PATCH] use password manager key from ESource auth-key property in ECal

---
 calendar/libecal/e-cal.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/calendar/libecal/e-cal.c b/calendar/libecal/e-cal.c
index a6de924..cf14118 100644
--- a/calendar/libecal/e-cal.c
+++ b/calendar/libecal/e-cal.c
@@ -1666,7 +1666,10 @@ open_calendar (ECal *ecal, gboolean only_if_exists, GError **error, ECalendarSta
 				e_source_peek_name (priv->source), username);
 
 		auth_type = e_source_get_duped_property (priv->source, "auth-type");
-		if (auth_type)
+		key = g_strdup(e_source_get_property (priv->source, "auth-key"));
+		if (key)
+			/* got key, do nothing */ ;
+		else if (auth_type)
 			key = build_pass_key (ecal);
 		else {
 			parent_user = e_source_get_duped_property (priv->source, "parent_id_name");
-- 
1.5.3.5.474-dirty

