From 8c69c7e63b25017f54e0ecf79980f9ce3ebd4215 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <ondrej.jirman@zonio.net>
Date: Thu, 25 Oct 2007 02:53:45 +0200
Subject: [PATCH] remove client in listener_died_cb only if reconnection failed

---
 calendar/libedata-cal/e-cal-backend.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/calendar/libedata-cal/e-cal-backend.c b/calendar/libedata-cal/e-cal-backend.c
index 1877288..344bcfd 100644
--- a/calendar/libedata-cal/e-cal-backend.c
+++ b/calendar/libedata-cal/e-cal-backend.c
@@ -362,7 +362,8 @@ listener_died_cb (gpointer cnx, gpointer data)
 {
 	EDataCal *cal = E_DATA_CAL (data);
 
-	e_cal_backend_remove_client (e_data_cal_get_backend (cal), cal);
+	if (ORBit_small_get_connection_status (e_data_cal_get_listener(cal)) == ORBIT_CONNECTION_DISCONNECTED)
+		e_cal_backend_remove_client (e_data_cal_get_backend (cal), cal);
 }
 
 static void
-- 
1.5.3.5.474-dirty

