From 0d744bedff672d0e27a2800195c401c347888796 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <ondrej.jirman@zonio.net>
Date: Tue, 13 Mar 2007 02:08:59 +0100
Subject: [PATCH] use password manager key from ESource auth-key property in ECal

---
 calendar/libecal/e-cal.c |   37 ++++++++++++++++++++-----------------
 1 files changed, 20 insertions(+), 17 deletions(-)

diff --git a/calendar/libecal/e-cal.c b/calendar/libecal/e-cal.c
index 9ac3bd9..92e5a9f 100644
--- a/calendar/libecal/e-cal.c
+++ b/calendar/libecal/e-cal.c
@@ -1658,23 +1658,26 @@ open_calendar (ECal *ecal, gboolean only_if_exists, GError **error, ECalendarSta
 		prompt = g_strdup_printf (_("Enter password for %s (user %s)"),
 				e_source_peek_name (priv->source), username);
 
-		auth_type = e_source_get_duped_property (priv->source, "auth-type");
-		if (auth_type) 
-			key = build_pass_key (ecal);
-		else {
-			parent_user = e_source_get_duped_property (priv->source, "parent_id_name");
-			if (parent_user) {
-				key = build_proxy_pass_key (ecal, parent_user);
-				/* 
-				   This password prompt will be prompted rarely. Since the key that is passed to 
-				   the auth_func corresponds to the parent user.
-				 */
-				prompt = g_strdup_printf (_("Enter password for %s to enable proxy for user %s"), e_source_peek_name (priv->source), parent_user);
-				g_free (parent_user);
-			} else 
-				key = g_strdup (e_cal_get_uri (ecal));
-		}
-		g_free (auth_type);
+		key = g_strdup(e_source_get_property (priv->source, "auth-key"));
+		if (!key) {
+      auth_type = e_source_get_duped_property (priv->source, "auth-type");
+      if (auth_type) 
+        key = build_pass_key (ecal);
+      else {
+        parent_user = e_source_get_duped_property (priv->source, "parent_id_name");
+        if (parent_user) {
+          key = build_proxy_pass_key (ecal, parent_user);
+          /* 
+             This password prompt will be prompted rarely. Since the key that is passed to 
+             the auth_func corresponds to the parent user.
+           */
+          prompt = g_strdup_printf (_("Enter password for %s to enable proxy for user %s"), e_source_peek_name (priv->source), parent_user);
+          g_free (parent_user);
+        } else 
+          key = g_strdup (e_cal_get_uri (ecal));
+      }
+      g_free (auth_type);
+    }
 
 		if (!key) {
 			e_calendar_remove_op (ecal, our_op);
-- 
1.5.3.2.99-dirty

