From cf316d2622aa8cf7d59ca370cb2211022200229d Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <ondrej.jirman@zonio.net>
Date: Fri, 12 Oct 2007 13:55:37 +0200
Subject: [PATCH] fix memory corruption in e_file_cache_get_keys()

---
 libedataserver/e-file-cache.c |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/libedataserver/e-file-cache.c b/libedataserver/e-file-cache.c
index 2299431..c64a4a0 100644
--- a/libedataserver/e-file-cache.c
+++ b/libedataserver/e-file-cache.c
@@ -346,6 +346,14 @@ e_file_cache_get_objects (EFileCache *cache)
 	return list;
 }
 
+static void
+add_key_to_slist (const char *key, const char *value, gpointer user_data)
+{
+	GSList **keys = user_data;
+
+	*keys = g_slist_append (*keys, (char *) key);
+}
+
 /**
  * e_file_cache_get_keys:
  */
@@ -359,7 +367,7 @@ e_file_cache_get_keys (EFileCache *cache)
 
 	priv = cache->priv;
 
-	e_xmlhash_foreach_key (priv->xml_hash, (EXmlHashFunc) add_key_to_list, &list);
+	e_xmlhash_foreach_key (priv->xml_hash, (EXmlHashFunc) add_key_to_slist, &list);
 
 	return list;
 }
-- 
1.5.3.4.206-dirty

