From 8e37b55dd0effd139791297fd6001d5add647539 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <ondrej.jirman@zonio.net>
Date: Fri, 12 Oct 2007 13:55:37 +0200
Subject: [PATCH] fix memory corruption in e_file_cache_get_keys()

---
 libedataserver/e-file-cache.c |   14 +++++++-------
 1 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/libedataserver/e-file-cache.c b/libedataserver/e-file-cache.c
index 2299431..856e30f 100644
--- a/libedataserver/e-file-cache.c
+++ b/libedataserver/e-file-cache.c
@@ -243,11 +243,11 @@ e_file_cache_remove (EFileCache *cache)
 }
 
 static void
-add_key_to_list (const char *key, const char *value, gpointer user_data)
+add_key_to_slist (const char *key, const char *value, gpointer user_data)
 {
-	GList **keys = user_data;
+	GSList **keys = user_data;
 
-	*keys = g_list_append (*keys, (char *) key);
+	*keys = g_slist_append (*keys, (char *) key);
 }
 
 /**
@@ -262,16 +262,16 @@ gboolean
 e_file_cache_clean (EFileCache *cache)
 {
 	EFileCachePrivate *priv;
-	GList *keys = NULL;
+	GSList *keys = NULL;
 
 	g_return_val_if_fail (E_IS_FILE_CACHE (cache), FALSE);
 
 	priv = cache->priv;
 
-	e_xmlhash_foreach_key (priv->xml_hash, (EXmlHashFunc) add_key_to_list, &keys);
+	e_xmlhash_foreach_key (priv->xml_hash, (EXmlHashFunc) add_key_to_slist, &keys);
 	while (keys != NULL) {
 		e_file_cache_remove_object (cache, (const char *) keys->data);
-		keys = g_list_remove (keys, keys->data);
+		keys = g_slist_remove (keys, keys->data);
 	}
 
 	return TRUE;
@@ -359,7 +359,7 @@ e_file_cache_get_keys (EFileCache *cache)
 
 	priv = cache->priv;
 
-	e_xmlhash_foreach_key (priv->xml_hash, (EXmlHashFunc) add_key_to_list, &list);
+	e_xmlhash_foreach_key (priv->xml_hash, (EXmlHashFunc) add_key_to_slist, &list);
 
 	return list;
 }
-- 
1.5.4-dirty

