From bb8902b1c47cb835d74ae9527851004611a65a38 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <ondrej.jirman@zonio.net>
Date: Sun, 14 Oct 2007 20:39:24 +0200
Subject: [PATCH] always pass username/password to e_cal_open

---
 calendar/libecal/e-cal.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/calendar/libecal/e-cal.c b/calendar/libecal/e-cal.c
index c3fa47e..78d66ea 100644
--- a/calendar/libecal/e-cal.c
+++ b/calendar/libecal/e-cal.c
@@ -1632,7 +1632,7 @@ open_calendar (ECal *ecal, gboolean only_if_exists, GError **error, ECalendarSta
 	g_mutex_unlock (priv->mutex);
 
 	/* see if the backend needs authentication */
-	if ( (priv->mode !=  CAL_MODE_LOCAL) && e_source_get_property (priv->source, "auth")) {
+	if (e_source_get_property (priv->source, "auth")) {
 		char *prompt, *key;
 		char *parent_user;
 
-- 
1.5.4-dirty

