From fe851b42aa12857d1e9983b9239b8f4c0b25169c Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <ondrej.jirman@zonio.net>
Date: Wed, 24 Oct 2007 21:08:04 +0200
Subject: [PATCH] don't leak ENameSelectorDialog in ENameSelector

---
 libedataserverui/e-name-selector-dialog.c |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/libedataserverui/e-name-selector-dialog.c b/libedataserverui/e-name-selector-dialog.c
index f2b244d..dd728fd 100644
--- a/libedataserverui/e-name-selector-dialog.c
+++ b/libedataserverui/e-name-selector-dialog.c
@@ -118,8 +118,11 @@ e_name_selector_dialog_populate_categories (ENameSelectorDialog *name_selector_d
 	category_option_menu = glade_xml_get_widget (name_selector_dialog->gui, "optionmenu-category");
 
 	/* Categories are already sorted */
-	category_list = e_categories_get_list () ;
-	category_list = g_list_prepend (category_list, _("Any Category"));
+	if (category_list == NULL)
+	{
+		category_list = e_categories_get_list () ;
+		category_list = g_list_prepend (category_list, _("Any Category"));
+	}
 
 	category_menu = gtk_menu_new ();
 	l = category_list;
@@ -310,7 +313,6 @@ e_name_selector_dialog_finalize (GObject *object)
 	g_array_free (name_selector_dialog->sections, TRUE);
 	g_object_unref (name_selector_dialog->source_list);
 	g_object_unref (name_selector_dialog->button_size_group);
-	g_list_free (category_list);
 
 	if (G_OBJECT_CLASS (e_name_selector_dialog_parent_class)->finalize)
 		G_OBJECT_CLASS (e_name_selector_dialog_parent_class)->finalize (object);
-- 
1.5.3.4.206-dirty

